lua/wikis/commons/Infobox/Extension/TeamHistoryAuto.lua

```diff
@@ -0,0 +1,501 @@
+	local _, teamText = self:_getTeamLinkAndText(transfer)
+
+	---@type Widget|string?
+	local role = transfer.role
+	if role then
+		local splitRole = mw.text.split(role --[[@as string]], ' ')
+		local roleData = ROLE_CONVERT[transfer.role:lower()] or ROLE_CONVERT[splitRole[#splitRole]:lower()]
+		if roleData.empty then
+			role = nil
+		else
+			role = roleData and Abbr(roleData) or transfer.role
+		end
+	end
+	if role == LOAN then
+		teamText = '&#8250;&nbsp;' .. teamText
+	end
+	---@type string|Widget
+	local teamDisplay = teamText
+	if role then
+		teamDisplay = Span{
+			css = {['padding-left'] = '3px', ['font-style'] = 'italic'},
+			children = {
+				'(',
+				role
+				')',
+			}
+		}
+	end
+
+	local positionIcon
+	if POSITION_ICON_DATA then
+		local position = (transfer.position or ''):lower()
+		positionIcon = (POSITION_ICON_DATA[position] or POSITION_ICON_DATA['']) .. '&nbsp;'
+	end
+
+	local leaveateDisplay = self:_buildLeaveDateDisplay(transfer)
+
+	if not self.config.hasHeaderAndRefs then
+		return Tr{children = {
+			Td{
+				classes = {'th-mono'},
+				css = {float = 'left', width = '50%', ['font-style'] = 'italic'},
+				children = {
+					transfer.joinDateDisplay,
+					leaveateDisplay and ' &#8212; ' or nil,
+					leaveateDisplay,
+				},
+			},
+			Td{
+				css = {float = 'right', width = '50%'},
+				children = WidgetUtil.collect(
+					positionIcon,
+					teamDisplay
+				),
+			},
+		}}
+	end
+
+	return Tr{children = {
+		Td{
+			classes = {'th-mono'},
+			css = {['white-space'] = 'nowrap', ['vertical-align'] = 'top'},
+			children = {
+				transfer.joinDateDisplay,
+				TeamHistoryAuto._displayRef(transfer.reference.join, transfer.joinDateDisplay)
+			},
+		},
+		Td{
+			classes = {'th-mono'},
+			css = {['white-space'] = 'nowrap', ['vertical-align'] = 'top', ['padding-left'] = '5px'},
+			children = {
+				leaveateDisplay,
+				TeamHistoryAuto._displayRef(transfer.reference.leave, transfer.leaveDateDisplay)
+			},
+		},
+		Td{
+			css = {['padding-left'] = '5px'},
+			children = WidgetUtil.collect(
+				positionIcon and (positionIcon .. '&nbsp;') or nil,
+				teamDisplay
+			),
+		},
+	}}
+end
+
+---@param references table[]
+---@param date string
+---@return Widget?
+function TeamHistoryAuto._displayRef(references, date)
+	local refs = Array.map(TransferRef.fromStorageData(references), function(reference)
+		return reference.link and TransferRef.useReference(reference, date) or nil
+	end)
+
+	if Logic.isEmpty(refs) then return end
+
+	return Fragment{children = {
+		Span{
+			css = {['font-size'] = '50%'},
+			children = {'&thinsp;'},
+		},
+		refs
+	}}
+end
+
+---@param transfer table
+---@return string|Widget?
+function TeamHistoryAuto:_buildLeaveDateDisplay(transfer)
+	if transfer.leaveDateDisplay then return transfer.leaveDateDisplay end
+
+	local lowerCasedRole = (transfer.role or ''):lower()
+	if lowerCasedRole == 'military' or not Table.includes(SPECIAL_ROLES_LOWER, (transfer.role or ''):lower()) then
```

Comment:
I don't understand the military special case? (Also I think it doesn't work live so this would be a change?)

Reply:
we ucfirst role in transfers, but it isn't really standardized ...